ci-droid-tasks-consumer-infrastructure/src/main/java/com/societegenerale/cidroid/tasks/consumer/infrastructure/FeignRemoteGitHub.java

```diff
@@ -149,19 +149,16 @@ default PullRequest createPullRequest(String repoFullName, PullRequestToCreate n
 
     @Override
     default void closePullRequest(String repoFullName, int prNumber) {
-        Map<String, String> body = new HashMap<>();
-        body.put("state", "closed");
```

Comment:
I used a map here to avoid creating a brand new class just for this use case. But if you're willing to do it (which is probably better), you should rather have a class – for example named PullRequestMetatdata – containing a "state" field. You can then set it to "closed" here and give the whole object to feign as body. The map was just here to simulate the json format, there's no need to keep it if there is now a java object.

Plus, if in the future there is the need to update other things about a PR (like the title for instance), it will be possible by just adding a "title" field to this object and calling the same updatePullRequest method.

Reply:
actually, the main thing I wanted to do here was to make the Map static to avoid creating a new one with same content every time. But because FeignRemoteGitHub is an interface, I couldn't create a static block.

Then I remembered that I could do it without a static block, with a one liner at declaration time as  mentioned here : https://stackoverflow.com/questions/7125536/when-would-i-use-java-collections-singletonmap-method 

which is even better, because the Map is immutable. I'm updating the PR with that change.